backend/src/api/routers/projects.py

```diff
@@ -0,0 +1,138 @@
+import uuid
+from typing import Annotated, AsyncIterator
+
+import pydantic
+from api.auth.users import current_active_user
+from database.session import async_session_maker
+from fastapi import APIRouter, Depends, HTTPException, status
+from models.project import Project
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+
+router = APIRouter()
+
+
+class ReadProject(pydantic.BaseModel):
+    id: uuid.UUID
+    title: str
+    summary: str | None = None
+    status: str
+
+
+class CreateProject(pydantic.BaseModel):
+    title: str
+    summary: str | None = None
+    status: str
+
+
+class UpdateProject(pydantic.BaseModel):
+    title: str | None = None
+    summary: str | None = None
+    status: str | None = None
+
+
+async def get_project(project_id: uuid.UUID, session: AsyncSession) -> Project:
```

Comment:
This function could be turned into a dependency:

```suggestion
async def get_project(project_id: Annotated[uuid.UUID, Path()], session: Annotated[AsyncSession, SessionWithTransactionContext]) -> Project:
```

Reply:
I actually had issues with that for when using unit of work patterns for ORM update and delete. The issue was that the return value then belonged to the injected session and I needed access to that session in the function that I inject `get_project` into. Otherwise I could not continue with the unit of work pattern.
Do you know a solution for this or is the unit of work pattern just not suitable here?

Either way I refactored the dependency injections independently to allow for better testing so the issue is not relevant anymore right now. But I would still be interested in how to solve it (Or whether it does not work with the unit of work pattern)